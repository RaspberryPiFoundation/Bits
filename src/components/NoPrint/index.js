import { Base } from 'react-iotacss'
import classnames from 'classnames'
import PropTypes from 'prop-types'
import React from 'react'

export const NoPrint = ({ children, className, ...props }) => {
  const classNames = classnames(className, 'u-no-print')

  return (
    <Base className={classNames} {...props}>
      {children}
    </Base>
  )
}

NoPrint.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
}

export default NoPrint
