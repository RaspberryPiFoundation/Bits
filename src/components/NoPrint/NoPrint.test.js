import { shallow } from 'enzyme'
import NoPrint from '../NoPrint'
import React from 'react'

let noPrint

const children = 'Whatever'
const className = 'some-extra-class-name'

describe('<Slice />', () => {
  describe('Renders...', () => {
    beforeEach(() => {
      noPrint = shallow(<NoPrint>{children}</NoPrint>)
    })

    it('without crashing', () => {
      expect(noPrint.find('.u-no-print')).toExist()
    })

    it('matches its snapshot', () => {
      expect(noPrint).toMatchSnapshot()
    })
  })

  describe('With classNames', () => {
    it('adds the correct className', () => {
      noPrint = shallow(<NoPrint className={className}>{children}</NoPrint>)
      expect(noPrint.find('.u-no-print')).toHaveClassName(className)
    })
  })
})
