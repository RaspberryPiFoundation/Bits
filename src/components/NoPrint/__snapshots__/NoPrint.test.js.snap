// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Slice /> Renders... matches its snapshot 1`] = `
<Base
  className="u-no-print"
  tagName="div"
>
  Whatever
</Base>
`;
